src/config.ts

```diff
@@ -1,25 +1,39 @@
 import { extname } from 'node:path';
+import { existsSync } from 'node:fs';
 import { readFile, writeFile } from 'node:fs/promises';
 import type { CSpellSettings } from 'cspell';
 import { searchForConfig } from 'cspell-lib';
+import findDefaultConfigPath from 'application-config-path';
 // eslint-disable-next-line import/no-relative-packages
 import { previousState } from './shared';
 
 let configPath: string | undefined;
 
-export const findConfig = async (config?: string) => {
+export const writeToSettings = async (settings: CSpellSettings | { cspell: CSpellSettings }) => {
+	await writeFile(configPath!, JSON.stringify(settings, null, 4));
+};
+
+export const findOrCreateConfig = async (config?: string) => {
 	// Try to locate a config file in the current working directory, or with the `config` option if it was provided.
 	const configSource = config ?? (await searchForConfig(process.cwd()))?.__importRef?.filename;
 
+	// LocalConfigPath is a path to a config file in the user's configuration directory (platform dependent).
+	const defaultConfigPath = findDefaultConfigPath('cspell.json');
+
 	// If no config file was found, use/create a config file in the user's configuration directory (platform dependent).
-	const path = configSource ?? (await import('application-config-path')).default('cspell.json');
```

Comment:
Sorry for just mentioning this now, but wasn't your reasoning for separating the into a separate variable so that you could compare it to path on line 33? But, couldn't you instead just check if `!configSource`?

Reply:
Like this?

```ts
export const findOrCreateConfig = async (config?: string) => {
	// Try to locate a config file in the current working directory, or with the `config` option if it was provided.
	const configSource = config ?? (await searchForConfig(process.cwd()))?.__importRef?.filename;

	// If no config file was found, use/create a config file in the user's configuration directory (platform dependent).
	const path = configSource ?? (await import('application-config-path')).default('cspell.json');

	// Only JSON files are supported to prevent more dependencies for yml parsing. If the config file is not a JSON
	// file, it can still be used, but it won't be updated with new ignored words.
	if (extname(path) === '.json') {
		configPath = path;
	}

	// If configSource is undefined, check if default config path exists. If it doesn't, create it.
	if (!configSource && !existsSync(path)) {
		await writeToSettings({});
	}

	return path;
};
```

Reply:
Yes, but you noted that a regular import is faster than a dynamic import, so you can import application-config-path regularly.

Reply:
Done :)